src/features/Planner/data/PlanApi.ts

```diff
@@ -15,11 +19,41 @@ const PlanApi = {
             newIds: r.data.newIds,
         })),
 
-    renameItem: (planId, body) =>
-        promiseFlux(axios.put(`/${planId}/rename`, body), (r) => ({
-            type: PlanActions.UPDATED,
-            data: r.data.info,
-        })),
+    renameItem: (planId, name) =>
+        promiseFlux(
+            client.mutate({
+                mutation: RENAME_PLAN_ITEM,
+                variables: {
+                    id: planId,
+                    name,
+                },
+            }),
+            (result: FetchResult<RenamePlanItemMutation>) => {
+                const planItem = result?.data?.planner?.rename;
+                if (!planItem) {
+                    // TODO
+                    return;
+                }
+                return {
+                    type: PlanActions.UPDATED,
+                    data: {
+                        id: planItem.id,
+                        name: planItem.name,
+                        notes: planItem.notes,
+                        status: planItem.status,
+                        parentId: planItem.parent?.id || null,
+                        aggregateId: planItem.aggregate?.id || null,
+                        quantity: planItem.quantity?.quantity || null,
+                        units: planItem.quantity?.units?.name || null,
+                        uomId: planItem.quantity?.units?.id || null,
+                        ingredientId: planItem.ingredient?.id || null,
+                        bucketId: planItem.bucket?.id || null,
+                        preparation: planItem.preparation,
+                    },
+                };
+            },
+            soakUpUnauthorized,
```

Comment:
This isn't the right approach for graphql - an unauthorized request will return a 200 with `errors` populated:

```
HTTP/1.1 200 OK
Date: Sat, 18 May 2024 18:35:25 GMT
Content-Type: application/json;charset=UTF-8
Content-Length: 247

{"errors":[{"message":"No user principal found. Are you logged in?","locations":[{"line":3,"column":5}],"path":["planner","plans"],"extensions":{"type":"NoUserPrincipalException","classification":"DataFetchingException"}}],"data":{"planner":null}}
```

Reply:
Yeah it's totally true - hence the TODO item at the top for "what happens when shit goes wrong". I wasn't sure how a promise rejection was going to work with graphql here yet.

Reply:
@barneyb I wrote this to just throw whatever error comes back since we handle auth errors at the root of the graphQL client. Just double-checking I am not missing something important here?

Reply:
I made a `throwAnyGraphQLErrors` function while doing the ... library stuff? If it goes up where that todo is, it'll at least not be silent in the face of failures.